shared/chat/conversation/messages/attachment/image/index.js

```diff
@@ -28,28 +28,59 @@ type Props = {
   onClick: () => void,
   onShowInFinder: null | (() => void),
   path: string,
+  fullPath: string,
   progress: number,
   progressLabel: string,
   showButton: null | 'play' | 'film',
   title: string,
   toggleMessageMenu: () => void,
   videoDuration: string,
+  inlineVideoPlayable: boolean,
   width: number,
 }
 
 type State = {
   loaded: boolean,
+  playingVideo: boolean,
 }
 
 class ImageAttachment extends React.PureComponent<Props, State> {
-  state = {loaded: false}
+  imageRef: any
+
+  constructor(props: Props) {
+    super(props)
+    this.imageRef = React.createRef()
+  }
+  state = {loaded: false, playingVideo: false}
   _setLoaded = () => this.setState({loaded: true})
+
+  _onClick = () => {
+    if (this.props.inlineVideoPlayable) {
+      this.imageRef.current.onVideoClick()
+      this.setState({playingVideo: !this.state.playingVideo})
```

Comment:
should use closure form
```
this.setState(p => ({playingVideo: !p.playingVideo}))
```

Reply:
Why?

Reply:
Oh because of `this`?

Reply:
no because setState is async so if your goal is the flip the flag you have to use this form